lib/filesystem/src/filesystem.cpp

```diff
@@ -559,25 +546,6 @@ namespace irods::experimental::filesystem::NAMESPACE_IMPL
         return p1_info.id == p2_info.id;
     }
 
-    auto data_object_size(rxComm& _comm, const path& _p) -> std::uintmax_t
-    {
-        const auto s = stat(_comm, _p);
-
-        if (s.error < 0) {
-            throw filesystem_error{"cannot get size", _p, make_error_code(s.error)};
-        }
-
-        if (s.type == UNKNOWN_OBJ_T) {
-            throw filesystem_error{"path does not exist", _p, make_error_code(OBJ_PATH_DOES_NOT_EXIST)};
-        }
-
-        if (s.type == DATA_OBJ_T) {
-            return static_cast<std::uintmax_t>(s.size);
-        }
-
-        return 0;
-    }
-
```

Comment:
this function's implementation will change in the future, so this can stay here for now.